### PR TITLE
test: add proptest + insta tests for unused dev-deps

### LINKED ISSUE #111: [super-qa] high: proptest and insta dev-dependencies never used
## Finding: Unused Testing Dependencies

- **ID:** H8 + H9
- **Severity:** high
- **Category:** testing
- **Location:** Cargo.toml (dev-dependencies)
- **Detected by:** test-engineer agent

## Description

Both \`proptest\` (property-based testing) and \`insta\` (snapshot testing) are declared as dev-dependencies but have zero usage in the crate.

### proptest candidates

1. \`serialize_embedding\`/\`deserialize_embedding\` roundtrip (arbitrary Vec<f32>, arbitrary dim)
2. \`cosine_similarity\` properties: symmetry, identical vectors = 1.0, bounded [-1,1]
3. \`rrf_merge\` properties: idempotent ranking, all input IDs appear in output
4. \`ScopeTree\` path resolution roundtrip
5. \`content_hash\` determinism

### insta candidates

1. \`EngineSnapshot\` JSON structure (dump format is a public API contract)
2. \`FactExplanation\` output
3. \`EngineStatistics\` output
4. \`BootstrapReport\` output

## Suggested Approach

Either add tests using these frameworks or remove the unused dependencies.

### PR DESCRIPTION
**Pull request by dutiona**
## Summary

- Adds **proptest** property-based tests for 5 candidates identified in #111:
  - `serialize_embedding`/`deserialize_embedding` roundtrip (arbitrary `Vec<f32>`, arbitrary dim)
  - `cosine_similarity` properties: symmetry, identical vectors → 1.0, bounded [-1,1]
  - `rrf_merge` properties: all input IDs appear in output, descending order, positive scores
  - `ScopeTree` path resolution roundtrip + ancestors always end at root
  - `content_hash` determinism + always 32 hex chars

- Adds **insta** snapshot tests for 4 candidates:
  - `EngineStatistics` (empty + populated engine)
  - `FactExplanation` (active fact)
  - `EngineSnapshot` (empty + populated, with timestamp/embedding redactions)
  - `BootstrapReport` (default + populated)

- Enables insta `redactions` feature for stable snapshot comparisons
- Adds `proptest-regressions/` to `.gitignore`

**Test count:** 420 → 439 (+19 new tests)

## Test plan

- [x] `cargo test --all-features` — 439 passed, 0 failed
- [x] `cargo clippy --all-targets --all-features` — no new warnings
- [x] `cargo fmt --check` — clean
- [x] Snapshot tests stable across multiple runs (HashMap ordering handled via `sorted_redaction`)
- [x] Proptest passes with bounded f32 range (avoids overflow-induced NaN)

Fixes #111

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/store/mod.rs [hunk 1]
```diff
@@ -117,4 +117,29 @@ mod tests {
...
         assert!(recovered.is_empty());
     }
+
+    mod proptest_embedding {
+        use super::*;
+        use proptest::prelude::*;
+
+        proptest! {
+            #[test]
+            fn roundtrip(embedding in proptest::collection::vec(any::<f32>(), 0..512)) {
```
[the remark this answers is not in the document]
Comment (dutiona) [reply src/store/mod.rs]: Fixed: added `prop_filter("NaN != NaN in PartialEq", |f| !f.is_nan())` to exclude NaN from the roundtrip test. The serialization itself is byte-level correct for NaN, but `PartialEq` for `f32` follows IEEE 754 where NaN != NaN.

File: src/scope/tree.rs [hunk 2]
```diff
@@ -283,4 +283,61 @@ mod tests {
...
+        }
+
+        proptest! {
+            #![proptest_config(ProptestConfig::with_cases(32))]
+
+            #[test]
+            fn resolve_roundtrip(path in scope_path(4)) {
+                let conn = open_memory().unwrap();
+                init_schema(&conn).unwrap();
+                migrate(&conn, None).unwrap();
```
[the remark this answers is not in the document]
Comment (dutiona) [reply src/scope/tree.rs:306]: Acknowledged. Each test module in this codebase has its own `setup()` helper (see `store/facts.rs`, `search/vector.rs`, `search/hybrid.rs`). The proptest module needs its own DB per test case anyway (proptest runs many iterations), so extracting across module boundaries would add coupling without benefit.

File: src/inspect/dump.rs [hunk 3]
```diff
@@ -455,4 +455,55 @@ mod tests {
...
         assert_eq!(&bytes[..4], &[0x28, 0xb5, 0x2f, 0xfd], "zstd magic bytes");
     }
+
+    #[test]
+    fn snapshot_empty_engine_dump() {
+        use crate::store::schema::{init_schema, migrate, open_memory};
+
+        let conn = open_memory().unwrap();
+        init_schema(&conn).unwrap();
+        migrate(&conn, None).unwrap();
```
[the remark this answers is not in the document]
Comment (dutiona) [reply src/inspect/dump.rs:465]: Acknowledged. Same rationale as above — the DB setup follows the existing pattern in `dump.rs` tests (see `streaming_output_matches_snapshot_schema` and `streaming_empty_database` which use the same pattern). Refactoring would affect pre-existing code beyond this PR scope.
